catalogs/catalog-common/build.gradle.kts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
+plugins {
+  id("java")
+  alias(libs.plugins.shadow)
+  `maven-publish`
+}
+
+dependencies {
+  implementation(project(":core"))
+  implementation(project(":catalogs:catalog-hive"))
+  implementation(project(":catalogs:catalog-lakehouse-iceberg"))
+  implementation(project(":catalogs:catalog-jdbc-mysql"))
+  implementation(project(":catalogs:catalog-jdbc-postgresql"))
+}
+
+tasks.test {
+  useJUnitPlatform()
+}
+
+tasks.withType<ShadowJar>(ShadowJar::class.java) {
+  isZip64 = true
+  configurations = listOf(project.configurations.runtimeClasspath.get())
+  archiveClassifier.set("")
+
+  dependencies {
+    exclude("org.*")
+    exclude("javax.*")
+  }
+
+  exclude("**/package-info.class")
+  exclude("**/*.properties")
+  exclude("**/*.html")
+  exclude("org/**")
+  exclude("META-INF/**")
+  exclude("module-info.class")
+  exclude("com/google/**")
+  exclude("com/fasterxml/**")
+  exclude("javax/**")
+  exclude("schema/**")
+  exclude("fr/**")
+  exclude("google/**")
+  exclude("groovy/**")
+  exclude("images/**")
+  exclude("**/*.conf")
+  exclude("**/*.so")
+  exclude("**/*.sxd")
+  exclude("**/*.ddl")
```

Comment:
Did you exclude everything and only leave something you wanted?

Reply:
Yes, the final jar is like this.

<img width="738" alt="image" src="https://github.com/datastrato/gravitino/assets/15794564/5e89c14b-6492-43cd-ae3c-cde1db002e38">